tensorflow/c/experimental/filesystem/filesystem_interface.h

```diff
@@ -84,22 +84,28 @@ typedef struct TF_TransactionToken {
 } TF_TransactionToken;
 
 typedef struct TF_Filesystem_Option_Value {
-  int type_tag;
-  int num_values;
+  int type_tag; // type of the value (int, real, buffer)
   union {
     int64_t inv_val;
     double real_val;
     struct {
       char* buf;
       int buf_length;
     } buffer_val;
-  } * values;  // owned
```

Comment:
Has to be a pointer type as we need to allow multiple values.

`value` is a better name than `u`

Reply:
It definitely works in C, but in C++ because `malloc` and so return `void*` it needs a cast.

It seems using `typeof` works: https://godbolt.org/z/6j31a14KK

Unfortunately, because the `struct` and `union` is in C-land, we cannot give a name to the union and reference it, afaik.

Reply:
It even works with `new`: https://godbolt.org/z/hfrseqfM5

Reply:
Oh, MSVC doesn't like that, so we have to extract the union outside (and then comment it): https://godbolt.org/z/T9MbhsnEx

Reply:
@mihaimaruseac  I didn't know we can use `typeof(v.values)`. 👍 

I have wrapped union outside to allow compiling with MSVC.